plugins/trivy/pkg/image/newest_images.go

```diff
@@ -54,22 +60,47 @@ func fetchTags(ctx context.Context, imageRepoName string) ([]string, error) {
 	return tags, nil
 }
 
-func filterAndSort(tags []string, currentTag string) []string {
-	newest := []string{}
-	c := version.NewConstrainGroupFromString(">" + currentTag)
-	currentTagSpecificToken := GetSpecificToken(currentTag)
-	for _, targetTag := range tags {
-		targetTagSpecificToken := GetSpecificToken(targetTag)
-		if c.Match(targetTag) && currentTagSpecificToken == targetTagSpecificToken {
-			newest = append(newest, targetTag)
+func filterAndSort(suggestedTags []string, curTagStr string) ([]string, error) {
+	curVersion, err := semver.NewVersion(curTagStr)
+	if err != nil {
+		logrus.Infof("could not parse current version %s: %v", curTagStr, err)
+		return []string{}, nil
+	}
+
+	constraint, err := semver.NewConstraint(">" + curTagStr)
+	if err != nil {
+		return nil, err
+	}
+
+	newest := []*semver.Version{}
+	for _, tag := range suggestedTags {
+		v, err := semver.NewVersion(tag)
+		if err != nil {
+			logrus.Infof("could not parse tag %s: %v", tag, err)
+			continue
+		}
+
+		cPre := curVersion.Prerelease()
+		vPre := v.Prerelease()
+		isKnownPreRelease := lo.Contains(knownPreReleases, cPre)
```

Comment:
I don't think we need to check `knownPreReleases` - just that the two match

Reply:
If we remove that, `v0.0.16-alpine` will not be recommended for `v0.0.14-beta1`

Reply:
fixed, I guess that was intended